### ISSUE TITLE
Lable style is not adopting theme style

### ISSUE THREAD

**Issue by Srj-Love** [Alexqwesa/app_bar_with_search_switch#4]
Hey there,

Thanks for making this package. This package is wonderful as my all requirement is in it.

- One issues I faced. when I'm entering text, my text style getting weird (bigger font).
- The controller is not getting disposed.

Here is my code.

```Scaffold(
              bottomNavigationBar: buildBottomBar(),
              appBar: AppBarWithSearchSwitch(
                bottom: buildPreferredSize(context),
                backgroundColor: context.theme.colorScheme.surface,
                fieldHintText: LocaleKeys.searchStudent.tr,
                searchInputDecoration: InputDecoration(
                  hintStyle: context.theme.textTheme.bodySmall,
                  labelStyle: context.theme.textTheme.bodySmall,
                  enabledBorder: InputBorder.none,
                  focusedBorder: InputBorder.none,
                  border: InputBorder.none,
                ),
                elevation: 0,
                keepAppBarColors: false,
                animation: AppBarAnimationSlideLeft.call,
                foregroundColor: context.theme.colorScheme.primary,
                onChanged: (text) {
                  debugPrint('onChanged...... $text');
                },
                onSubmitted: (text) => debugPrint('onSubmitted...... $text'),
                appBarBuilder: (context) {
                  return AppBar(
                    title: Text(LocaleKeys.attendance.tr),
                    titleTextStyle: context.textTheme.titleMedium,

                    bottom: buildPreferredSize(context),
                    actions:  const [
                      AppBarSearchButton(
                        buttonHasTwoStates: false,
                        searchActiveButtonColor: Colors.white,
                      ),
                    ],
                  );
                },
              ),
              body: TabBarView(
                controller: controller.tabController,
                physics: const NeverScrollableScrollPhysics(),
                // physics: controller.isMainTeacher.isFalse?const NeverScrollableScrollPhysics():const PageScrollPhysics(),
                children: const [

                  // needed
                  NeededAttendanceView(),

                  // inbox
                  InboxView(),

                  OutboxView(),

                  DoneAttendanceView(),
                ],
              )
          ),```

https://github.com/Alexqwesa/app_bar_with_search_switch/assets/21064334/e763dcb0-34b4-4424-bd0c-781e59d725bc

**Comment by Alexqwesa** [comment]
> * One issues I faced. when I'm entering text, my text style getting weird (bigger font).

Well, i think it's because by default it use:
```dart
          style: theme.textTheme.headlineSmall?.copyWith(
            color: mainWidget.keepAppBarColors
                ? theme.appBarTheme.foregroundColor ?? theme.canvasColor
                : theme.textTheme.headlineSmall?.color,
          ),
```
I am trying to use reasonable default values and also provide full ability to customize, so
in this case you could have used your own `title` like this:
```dart
AppBarWithSearchSwitch(
  title: (context) {
    final mainWidget = AppBarWithSearchSwitch.of(context)!;

    return TextField(
    ... // here should be your style and decorator!!!

          onSubmitted: mainWidget.submitCallbackForTextField,
          keyboardType: mainWidget.keyboardType,
          autofocus: true,
          controller: mainWidget.textEditingController,
```

I will think about addition of new parameter, something like TextFieldStyle?  or maybe I should have used theme.appBarTheme.titleTextStyle ? ...

**Comment by Alexqwesa** [comment]
> * The controller is not getting disposed.

What do you mean? textEditingController ?
Yes, it is static global one, what problem you have with this? 
If you want to control it - use `customTextEditingController` - but you will need to manually create/dispose it, 
anyway it would be nice if you provide your usecase(what you trying to do and what happen).

I was thinking about some way to use different controllers for each instance of `AppBarWithSearchSwitch`:
but it will required to have a way to distinguish between instances - maybe assign distinct controllers for each `AppBarWithSearchSwitch` with different `Key` parameter?

**Comment by Alexqwesa** [comment]
> Thanks for making this package.

Thanks))

**Comment by Alexqwesa** [comment]
I decide not to introduce new parameter and just use exist titleTextStyle, because by default, in search mode, title only contains TextField... and if you use custom title: you can use any style you want...
```
v1.6.0
[AppBarWithSearchSwitch].titleTextStyle now also applied to default TextField (in search mode),
```

**Comment by Alexqwesa** [comment]
it's been one month, with no feedback...

> One issues I faced. when I'm entering text, my text style getting weird (bigger font).

was fixed 3 week ago (v1.6.0)

> The controller is not getting disposed.

still waiting for clarifications...  (btw, currently textController and notifiers are global singletons by design)

please, open this bug report again if you still have issues or can add clarifications.
